ibm/resource_ibm_cis_waf_package.go

```diff
@@ -0,0 +1,164 @@
+			},
+			cisWAFPackageActionMode: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "WAF pakcage action mode",
+				ValidateFunc: InvokeValidator(
+					ibmCISWAFPackage, cisWAFPackageActionMode),
+			},
+			cisWAFPackageDescription: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "WAF package description",
+			},
+		},
+	}
+}
+
+func resourceIBMCISWAFPackageValidator() *ResourceValidator {
+
+	validateSchema := make([]ValidateSchema, 1)
+	sesitivity := "high, medium, low, off"
+	actionMode := "simulate, block, challenge"
+
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 cisWAFPackageSensitivity,
+			ValidateFunctionIdentifier: ValidateAllowedStringValue,
+			Type:                       TypeString,
+			Required:                   true,
+			AllowedValues:              sesitivity})
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 cisWAFPackageActionMode,
+			ValidateFunctionIdentifier: ValidateAllowedStringValue,
+			Type:                       TypeString,
+			Required:                   true,
+			AllowedValues:              actionMode})
+	ibmCISWAFPackageValidator := ResourceValidator{ResourceName: ibmCISWAFPackage, Schema: validateSchema}
+	return &ibmCISWAFPackageValidator
+}
+
+func resourceIBMCISWAFPackageUpdate(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisWAFPackageClientSession()
+	if err != nil {
+		return err
+	}
+
+	crn := d.Get(cisID).(string)
+	zoneID, _, err := convertTftoCisTwoVar(d.Get(cisDomainID).(string))
+	packageID, _, _, err := convertTfToCisThreeVar(d.Get(cisWAFPackageID).(string))
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneID = core.StringPtr(zoneID)
+
+	if d.HasChange(cisWAFPackageSensitivity) ||
+		d.HasChange(cisWAFPackageActionMode) {
+		opt := cisClient.NewUpdateWafPackageOptions(packageID)
+		if v, ok := d.GetOk(cisWAFPackageSensitivity); ok {
+			opt.SetSensitivity(v.(string))
+		}
+		if v, ok := d.GetOk(cisWAFPackageActionMode); ok {
+			opt.SetActionMode(v.(string))
+		}
+		result, response, err := cisClient.UpdateWafPackage(opt)
+		if err != nil {
+			log.Printf("Update waf package setting failed: %v", response)
+			return err
+		}
+		d.SetId(convertCisToTfThreeVar(*result.Result.ID, zoneID, crn))
+	}
+
+	return resourceIBMCISWAFPackageRead(d, meta)
+}
+
+func resourceIBMCISWAFPackageRead(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisWAFPackageClientSession()
+	if err != nil {
+		return err
+	}
+	packageID, zoneID, crn, err := convertTfToCisThreeVar(d.Id())
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneID = core.StringPtr(zoneID)
+	opt := cisClient.NewGetWafPackageOptions(packageID)
+	result, response, err := cisClient.GetWafPackage(opt)
+	if err != nil {
+		log.Printf("Get waf package setting failed: %v", response)
```

Comment:
Since we don't have exists method can we check for 404 condition and set the 
```
if response != nil && response.StatusCode == 404 {
			d.SetId("")
			return nil
		}
```

Reply:
added them.